airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/SnowflakeDestination.kt

```diff
@@ -51,38 +62,108 @@ constructor(
 
     override fun check(config: JsonNode): AirbyteConnectionStatus? {
-        val snowflakeInternalStagingSqlOperations =
-            SnowflakeInternalStagingSqlOperations(nameTransformer)
         val dataSource = getDataSource(config)
         try {
+            val retentionPeriodDays = 1
+            val sqlGenerator = SnowflakeSqlGenerator(retentionPeriodDays)
             val database = getDatabase(dataSource)
-            val outputSchema = nameTransformer.getIdentifier(config["schema"].asText())
-            JdbcCheckOperations.attemptTableOperations(
-                outputSchema,
-                database,
-                nameTransformer,
-                snowflakeInternalStagingSqlOperations,
-                true,
-            )
-            attemptStageOperations(
-                outputSchema,
-                database,
-                nameTransformer,
-                snowflakeInternalStagingSqlOperations
+            val databaseName = config[JdbcUtils.DATABASE_KEY].asText()
+            val outputSchema = nameTransformer.getIdentifier(config[JdbcUtils.SCHEMA_KEY].asText())
+            val rawTableSchemaName: String =
+                if (getRawNamespaceOverride(RAW_SCHEMA_OVERRIDE).isPresent) {
+                    getRawNamespaceOverride(RAW_SCHEMA_OVERRIDE).get()
+                } else {
+                    JavaBaseConstants.DEFAULT_AIRBYTE_INTERNAL_NAMESPACE
+                }
+            val finalTableName =
+                nameTransformer.getIdentifier(
+                    "_airbyte_connection_test_" +
+                        UUID.randomUUID().toString().replace("-".toRegex(), "")
+                )
+            val snowflakeDestinationHandler =
+                SnowflakeDestinationHandler(databaseName, database, rawTableSchemaName)
+            val snowflakeStagingClient = SnowflakeStagingClient(database)
+            val snowflakeStorageOperation =
+                SnowflakeStorageOperation(
+                    sqlGenerator = sqlGenerator,
+                    destinationHandler = snowflakeDestinationHandler,
+                    retentionPeriodDays,
+                    snowflakeStagingClient
+                )
+            val streamId =
+                sqlGenerator.buildStreamId(outputSchema, finalTableName, rawTableSchemaName)
+            val streamConfig =
+                StreamConfig(
+                    id = streamId,
+                    destinationSyncMode = DestinationSyncMode.OVERWRITE,
+                    primaryKey = listOf(),
+                    cursor = Optional.empty(),
+                    columns = linkedMapOf(),
+                    generationId = 0,
+                    minimumGenerationId = 0,
+                    syncId = 0
+                )
+            // None of the fields in destination initial status matter
+            // for a dummy sync with type-dedupe disabled. We only look at these
```

Comment:
constructor injection in StreamOperation. Maybe i should use named param (easier for reviews)

Reply:
ahhhh. yeah, naked boolean parameters are tough for review. Intellij at least makes reading the code easier since it adds the param name automatically though 🤔 
![image](https://github.com/airbytehq/airbyte/assets/5741425/6a9705cb-11cd-46a6-831f-decb28617520)

so... your call, I don't feel strongly either way

Reply:
made it named.